web/olga/analytics/views.py

```diff
@@ -208,33 +298,28 @@ def create_instance_data(self, received_data, access_token):
             access_token (unicode): Secret key to allow edX instance send a data to server.
                                     If token is empty, it will be generated with uuid.UUID in string format.
         """
-        active_students_amount_day = int(received_data.get('active_students_amount_day'))
-        active_students_amount_week = int(received_data.get('active_students_amount_week'))
-        active_students_amount_month = int(received_data.get('active_students_amount_month'))
-        courses_amount = int(received_data.get('courses_amount'))
-        statistics_level = received_data.get('statistics_level')
-
-        stats = {
-            'active_students_amount_day': active_students_amount_day,
-            'active_students_amount_week': active_students_amount_week,
-            'active_students_amount_month': active_students_amount_month,
-            'courses_amount': courses_amount,
-            'statistics_level': statistics_level
-        }
-
-        edx_installation_object = EdxInstallation.objects.get(access_token=access_token)
-
-        if statistics_level == 'enthusiast':
-            self.extend_stats_to_enthusiast(received_data, stats, edx_installation_object)
-
-        previous_stats = InstallationStatistics.get_stats_for_this_day(edx_installation_object)
+        if statistics_date is None:
+            statistics_date = datetime.datetime.now()
+        
+        stats_to_write = dict((key, value) for key, value in stats.iteritems() if key in STATS_TO_WRITE_KEYS)
+        statistics_date.replace(hour=0, minute=0, second=0, microsecond=0, tzinfo=pytz.utc)
```

Comment:
tzinfo is redundant param, or I miss something?

Reply:
no, it's needed to make this datetime the tz-aware one

Reply:
removed